test/metabase/test/data/dataset_definitions.clj

```diff
@@ -1,36 +1,72 @@
 (ns metabase.test.data.dataset-definitions
   "Definitions of various datasets for use in tests with `with-temp-db`."
   (:require [clojure.tools.reader.edn :as edn]
-            [metabase.test.data.interface :refer [def-database-definition]]))
-
+            [metabase.test.data.interface :as di])
+  (:import java.sql.Time
+           java.util.Calendar))
 
 ;; ## Datasets
 
-(def ^:private ^:const edn-definitions-dir "./test/metabase/test/data/dataset_definitions/")
-
-;; TODO - move this to interface
-;; TODO - make rows be lazily loadable for DB definitions from a file
-(defmacro ^:private def-database-definition-edn [dbname]
-  `(def-database-definition ~dbname
-     ~@(edn/read-string (slurp (str edn-definitions-dir (name dbname) ".edn")))))
-
 ;; The O.G. "Test Database" dataset
-(def-database-definition-edn test-data)
+(di/def-database-definition-edn test-data)
 
 ;; Times when the Toucan cried
-(def-database-definition-edn sad-toucan-incidents)
+(di/def-database-definition-edn sad-toucan-incidents)
 
 ;; Places, times, and circumstances where Tupac was sighted
-(def-database-definition-edn tupac-sightings)
+(di/def-database-definition-edn tupac-sightings)
 
-(def-database-definition-edn geographical-tips)
+(di/def-database-definition-edn geographical-tips)
 
 ;; A very tiny dataset with a list of places and a booleans
-(def-database-definition-edn places-cam-likes)
+(di/def-database-definition-edn places-cam-likes)
 
 ;; A small dataset with users and a set of messages between them. Each message has *2* foreign keys to user --
 ;; sender and reciever -- allowing us to test situations where multiple joins for a *single* table should occur.
-(def-database-definition-edn avian-singles)
+(di/def-database-definition-edn avian-singles)
+
+(defn- date-only
+  "This function emulates a date only field as it would come from the
+  JDBC driver. The hour/minute/second/millisecond fields should be 0s"
+  [date]
+  (let [orig-cal (doto (Calendar/getInstance)
+                   (.setTime date))]
+    (-> (doto (Calendar/getInstance)
+          (.clear)
+          (.set Calendar/YEAR (.get orig-cal Calendar/YEAR))
+          (.set Calendar/MONTH (.get orig-cal Calendar/MONTH))
+          (.set Calendar/DAY_OF_MONTH (.get orig-cal Calendar/DAY_OF_MONTH)))
+        .getTime)))
+
+(defn- time-only
+  "This function will return a java.sql.Time object. To create a Time
```

Comment:
This isn't important but something I learned about recently that you might find useful:

I've just started using `column-enforce-mode` in Emacs. It's a minor mode that font locks text past a certain column (defined by `column-enforce-column`) as red. Maybe worth looking into.

It looks like these lines are all about ~70 characters wide, so I'm guessing you're trying to keep them less than the classic 80 chars, and since they're a little under 80 than that I'm guessing you're eyeballing them. So maybe it's a tool that you might find useful to make your life a little easier

Reply:
Do you have a preference on line length? I'm betting what I did here is write a big docstring, then M-q `paredit-reindent-defun`. I'm usually a fan of 120 chars, but I haven't changed the paredit defaults.

Reply:
Yeah, 120 is a good number. It seems like GitHub's limit is 120 characters before they add horizontal scrollbars to code, but with the `+` at the beginning of a PR I've figured out that you can only actually do 119 characters before GH cuts it off. So I have my column limit set to 118 just to be safe.

I didn't know about `paredit-reindent-defun`, I'll have to check that out. I've just been setting up a keyboard macro to do re-indentation manually like a chump 😓

Reply:
I think I've found a way to automate this. if you set the vars below:
``` emacs-lisp
 ;; Anywhere
  (setq clojure-docstring-fill-column 118)
  ;; In the clojure-mode-hook
   (set-fill-column 118)
```
The first will set clojure docstrings correctly at 118 and the second one will set code comments to wrap at 118, then you can just use the built in emacs fill-paragraph stuff that M-q uses.

@sbelak You might be interested in this too.